feat(manager/pixi): extract conda/pypi packages (#34646)
renovatebot/renovate: lib/modules/manager/pixi/readme.md

### Pixi package manager

Whenever the pixi config in `pyproject.toml` or `pixi.toml` file is updated, the `pixi.lock` file will be checked for updates as well.

> Currently support channels from anaconda and prefix.dev, other channels is not suppoted yet.
> 

Reviewer: ```suggestion
Currently support channels from anaconda and prefix.dev, other channels are not supported yet.

```

Renovatebot will pick pixi based on known lock file version and pixi version mapping,
optionally you can set `constraints.pixi` in your config to override this.

> The minimal support version of pixi is `0.40.0`, when you are setting `constraints.pixi` in your config, please be carefully.

Reviewer: Would be good to maybe mention why users need to be careful in that case :)

Author: this should be clear enough? if not would you mind help rephrase it?

> The minimal support version of pixi is `0.40.0` (old versions of pixi doesn't have command to generate lock file without installing the whole project), when you are setting `constraints.pixi` in your config, please be carefully to pick support pixi versions. Renovatebot must use a pixi `>=0.40.0` to generate lockfile.

Reviewer: ```suggestion
The minimal support version of pixi is `0.40.0` (`pixi lock` has only been introduced with that version).
Therefore when you are setting `constraints.pixi` in your config, please be careful to pick a supported pixi version.
```

### Pixi package manager

Currently support channels from anaconda and prefix.dev, other channels are not supported yet.

Whenever the pixi config in `pyproject.toml` or `pixi.toml` file is updated, `pixi.lock` file will be checked for updates as well.

> Renovatebot will pick pixi based on known lock file version and pixi version mapping to avoid upgrade the version of lock file,
> optionally you can set `constraints.pixi` in your config to override this.

Reviewer: ```suggestion
Renovatebot will pick a pixi version based on known mapping between lockfile version and pixi version.
This is done in order to avoid upgrading the lock file version.
Optionally you can set `constraints.pixi` in your config to override this.
```

### Pixi package manager

Currently support channels from anaconda and prefix.dev, other channels are not supported yet.

Whenever the pixi config in `pyproject.toml` or `pixi.toml` file is updated, `pixi.lock` file will be checked for updates as well.

Renovatebot will pick a pixi version based on known mapping between lockfile version and pixi version.
This is done in order to avoid upgrading the lock file version.
Optionally you can set `constraints.pixi` in your config to override this.

The minimal support version of pixi is `0.40.0` (`pixi lock` has only been introduced with that version).
Therefore when you are setting `constraints.pixi` in your config, please be careful to pick a supported pixi version.

This also means if you have a lock file with version `<6`, renovatebot can't generate lock file with same version, it will upgrade lock file to latest version.

If you want to map a mirror repo to official repo, you can use `registryAliases`:

for pixi.toml:

```toml
[project]
name = "example-project"
authors = []
channels = ["https://mirrors.of.some.company/conda/conda-forge"]
platforms = ["win-64"]
version = "0.1.0"
```

```json
{
  "registryAliases": {

>     "https://mirrors.of.some.company/conda/conda-forge": "conda-forge"

Reviewer 2: ```suggestion
    "mirrors.of.some.company/conda/conda-forge": "conda-forge"
```

Author: I'm not sure about this... Is `registryAliases` used outside manager so it should not contain url scheme??

https://github.com/renovatebot/renovate/blob/a4f41929086ae477665e67d4ead6d7f08f6c6185/lib/modules/manager/pixi/extract.ts#L145-L151

Author: OK I find pixi has native support for registry aliases, so I'll remove it from here
